WooCommerce/Classes/ViewRelated/Orders/Order Filters/Order Status Filter/OrderStatusFilterViewController.swift

```diff
@@ -0,0 +1,206 @@
+        guard let status = row.status else {
+            return
+        }
+        if selected.contains(status) {
+            selected.removeAll { $0 == status }
+        }
+        else {
+            selected.append(status)
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension OrderStatusFilterViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension OrderStatusFilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row] {
+        case .any:
+            selected = []
+            onCompletion(selected)
+            tableView.reloadData()
+        default:
+            selectOrDelesectRow(rows[indexPath.row])
+            onCompletion(selected)
+            tableView.reloadData()
+            return
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+}
+
+// MARK: - Cell configuration
+//
+private extension OrderStatusFilterViewController {
+    enum Row: CaseIterable {
+        case any
+        case pending
+        case processing
+        case onHold
+        case failed
+        case cancelled
+        case completed
+        case refunded
+
+        var status: OrderStatusEnum? {
+            switch self {
+            case .any:
+                return nil
+            case .pending:
+                return .pending
+            case .processing:
+                return .processing
+            case .onHold:
+                return .onHold
+            case .failed:
+                return .failed
+            case .cancelled:
+                return .cancelled
+            case .completed:
+                return .completed
+            case .refunded:
+                return .refunded
+            }
+        }
+
+        var type: UITableViewCell.Type {
+            BasicTableViewCell.self
+        }
+
+        var reuseIdentifier: String {
+            type.reuseIdentifier
+        }
+    }
+
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as BasicTableViewCell:
+            configureStatus(cell: cell, row: row)
+        default:
+            fatalError()
```

Comment:
Maybe we should just leave the method silently in this case and replace the `fatalError` with an `assertionFailure` instead?